rfcs/0136-stabilize-incrementally.md

```diff
@@ -0,0 +1,199 @@
+
+## A plan all sides can be happy with
+
+Stabilizing the new CLI and Flakes will end the saga of the past few years.
+It is a last good chance to sooth some of these tensions and put us on a good foot moving forward.
+The new NixOS Foundation gives me hope Nix is serious about addressing these sorts of governance issues, and thus would be a good way to further demonstrate we are turning a new leaf.
+
+The plan below attempts to make all sides happy, despite their seemingly irreconcilable differences.
+The basic thrust is to proceed with stabilization in small steps.
+Two benefits are envisioned:
+
+### Benefit 1: Keep discussions on track.
+
+Threads on Flakes in the past have been impossible to follow because the sheer magnitude of different topics being discussed.
+Small stabilization steps are meant to yield *focused, bounded* discussions that actually read well.
+This also ensures any issues that do arise are actually brought to attention, and not lost within a deluge of other topics.
+
+### Benefit 2: Build trust by starting with the least controversial material.
+
+Flakes are obviously the most controversial, and so they are saved for last.
+The CLI is more broadly popular, but still is a lot of material to discuss.
+The store-only subcommands are Nix's "plumbing" as opposed to "porcelain" commands, and thus have the simplest (if not most user-friendly) purposes.
+This "dusty corner" off the new CLI is rather calm with a very constrained design space, and far less acrimony.
+
+## Conclusion
+
+By starting with these relatively easy material to stabilize, we can prove we can all come together as a community to agree on a design after all.
+This should build trust between opposing individuals and factions, giving us a foundation upon which to tackle the more challenging material in subsequent steps.
+
+Since these features became very popular while they are still unstable, there is now both an urgency to stabilize them, and little ability to modify them.
+This is the opposite of how experimental features are supposed to leisurely incubate with both plenty of flexibility to change them, and little rush to stabilize them until they feel ready.
+But, there is little we can do about this at this point, an this RFC recognizes that fact and does *not* try to resist it.
+
+But, we have crossed that Rubicon and there is no turning back; this RFC *doesn't* attempt to change Flakes or the command line 
+
+# Detailed design
+[design]: #detailed-design
+
+Step 1 is technical work, with a self-imposed deadline so we can be sure it doesn't delay stabilization too long.
+The remaining steps are stabilization steps.
+For each of them, a separate RFC or other discussion medium will describe the new interfaces to be stabilized, and solicit feedback.
+
+## Step 0: Stabilize the store-only installable-free CLI
+
+There are certain commands like `nix store gc`, `nix store ping` that do not take any positional arguments.
```

Comment:
Is it in scope for the RFC to define the exact set of these commands (either by enumerating them, or by stating which property they need to have)? If not, where would this be done?

Reply:
I am not sure whether it is in-scope, but here is one such (WIP) list https://github.com/NixOS/nix/issues/7701

Reply:
Out of scope is better, thanks